kolibri/core/analytics/utils.py

```diff
@@ -0,0 +1,115 @@
+import base64
+import datetime
+import hashlib
+import json
+
+from django.db.models import Count
+from django.db.models import Max
+from django.db.models import Min
+from django.db.models import Sum
+
+from kolibri.core.auth.constants import role_kinds
+from kolibri.core.auth.models import FacilityUser
+from kolibri.core.content.models import LocalFile
+from kolibri.core.exams.models import Exam
+from kolibri.core.lessons.models import Lesson
+from kolibri.core.logger.models import AttemptLog
+from kolibri.core.logger.models import ContentSessionLog
+from kolibri.core.logger.models import ContentSummaryLog
+from kolibri.core.logger.models import ExamAttemptLog
+from kolibri.core.logger.models import ExamLog
+from kolibri.core.logger.models import UserSessionLog
+
+
+def dump_zipped_json(data):
+    jsondata = json.dumps(data)
+    try:
+        # perform the import in here as zlib isn't available on some platforms
+        import zlib
+        jsondata = zlib.compress(jsondata)
+    except:  # noqa
+        pass
+    return jsondata
+
+
+def extract_facility_statistics(facility):
+
+    dataset_id = facility.dataset_id
+
+    learners = FacilityUser.objects.filter(dataset_id=dataset_id).exclude(roles__kind__in=[role_kinds.ADMIN, role_kinds.COACH])
+    coaches = FacilityUser.objects.filter(dataset_id=dataset_id, roles__kind__in=[role_kinds.ADMIN, role_kinds.COACH])
+
+    usersessions = UserSessionLog.objects.filter(dataset_id=dataset_id)
+    contsessions = ContentSessionLog.objects.filter(dataset_id=dataset_id, time_spent__lt=3600 * 2)
+
+    usersess_agg = (usersessions
```

Comment:
perhaps a comment explaining that the aggregates below are used to calculate the first time this device has been interacted with and the last time it has been interacted with?